src/table/dtable.jl

```diff
@@ -0,0 +1,110 @@
+import Tables
+import TableOperations
+
+import Base:fetch
+
+export DTable
+
+const VTYPE = Vector{Union{Dagger.Chunk,Dagger.EagerThunk}}
+
+"""
+    struct DTable
+
+Structure representing the distributed table based on Dagger.
+
+The table is stored as a vector of `Chunk` structures which hold partitions of the table.
+That vector can also store `EagerThunk` structures when an operation that modifies
+the underlying partitions was applied to it (currently only `filter`).
+
+Underlying partitions should always be `DataFrame` structures.
+"""
+struct DTable
+    chunks::VTYPE
+    tabletype
+    DTable(chunks::VTYPE, tabletype) = new(chunks, tabletype)
+    DTable(chunks::Vector{Dagger.EagerThunk}, args...; kwargs...) = new(VTYPE(chunks), args...; kwargs...)
+    DTable(chunks::Vector{Dagger.Chunk}, args...; kwargs...) = new(VTYPE(chunks), args...; kwargs...)
```

Comment:
```suggestion
    DTable(chunks::Vector{Dagger.EagerThunk}, args...; kwargs...) = DTable(VTYPE(chunks), args...; kwargs...)
    DTable(chunks::Vector{Dagger.Chunk}, args...; kwargs...) = DTable(VTYPE(chunks), args...; kwargs...)
```

Internal ctors don't take kwargs, so we need to forward to external ctors instead. Maybe move these out of the struct body?